src/builtin.c

```diff
@@ -657,6 +657,39 @@ static jv f_format(jq_state *jq, jv input, jv fmt) {
     }
     jv_free(input);
     return line;
+  } else if (!strcmp(fmt_s, "urid")) {
+    jv_free(fmt);
+    input = f_tostring(jq, input);
+
+    jv line = jv_string("");
+    char *s = (char *)jv_string_value(input);
+    const char *errmsg = "is not a valid uri encoding";
+    while (*s) {
+      if (*s != '%') {
+        line = jv_string_append_buf(line, s++, 1);
+      } else {
+        char unicode[4] = {0};
```

Comment:
I don't think it makes a difference since you are only checking `&1` here, but I would declare this `unsigned char` to avoid problems since you are using `>>`